### PR TITLE
rpc,circuitbreaker,stop,testutils,nodedialer: small extracts from #99191

### PR DESCRIPTION
**Pull request by tbg**
See individual commits.

- nodedialer: up test heartbeat timeout to 500ms: this allowed stressing the tests "harder".
- testutils: print error source in SucceedsSoon: this was useful for understanding failures.
- stop: add OnQuiesce: this is used in #99191.
- breaker: allow testing override of Signal: this is used in #99191.
- rpc: add GetOffset method: this is used in #99191.
- rpc: improve ManualHeartbeatService: this is used in #99191.
- rpc: improve comment on onlyOnceDialer: the previous comment didn't reveal the reason for why this exists.

Epic: none
Release note: none

### CODE REVIEW

File: pkg/util/circuit/options.go [hunk 1]
```diff
@@ -41,7 +41,21 @@ type Options struct {
 	// not seeing any usage.
 	AsyncProbe func(report func(error), done func())
 
+	// SignalInterceptor gets to see and change the return value of the Signal
+	// method. This is useful for testing, as it allows "pretending" that the
+	// breaker is tripped without having to replace the probe method and
+	// coordinating with a running probe.
+	SignalInterceptor func(Signal) Signal
```
Comment (aliher1911) [inline pkg/util/circuit/options.go:48]: Is this field stale? We have another one which is accessed via test only methods. Can't see any uses for this. If we use it in future then we should have some comment explaining what is the difference.

File: pkg/util/stop/stopper.go [hunk 2]
```diff
@@ -240,6 +241,17 @@ func (s *Stopper) refuseRLocked() bool {
 	return s.mu.stopping || s.mu.quiescing
 }
 
+// OnQuiesce is like AddCloser, but invokes on quiesce.
```
Comment (aliher1911) [inline pkg/util/stop/stopper.go:244]: This needs a comment similar to `AddCloser` which says that f is called synchronously if stopper is already quiescing/stopping.

**Review by aliher1911 — APPROVED** [review]
With some nits.

File: pkg/util/circuit/options.go (hunk 1 above)
Comment (tbg) [reply pkg/util/circuit/options.go:48]: This was dead code - thanks for spotting. There is an unexported copy of this below in the struct, which is what is actually being used.

**Comment by tbg** [comment]
TFTR!

bors r=aliher1911

**Pull request merged**